packages/lodestar/src/sync/backfill/verify.ts

```diff
@@ -0,0 +1,24 @@
+import {verifyBlockSignature} from "@chainsafe/lodestar-beacon-state-transition";
+import {IBeaconConfig} from "@chainsafe/lodestar-config";
+import {BeaconState, SignedBeaconBlock} from "@chainsafe/lodestar-types/lib/allForks";
+
+export function verifyBlocks(config: IBeaconConfig, state: BeaconState, blocks: SignedBeaconBlock[]): void {
+  if (blocks.length === 0) {
+    return;
+  }
+  let lastBlock: SignedBeaconBlock | undefined;
+  for (const block of blocks) {
+    if (
+      lastBlock &&
+      !config.types.Root.equals(
+        config.getForkTypes(lastBlock.message.slot).BeaconBlock.hashTreeRoot(lastBlock.message),
+        block.message.parentRoot
+      )
+    ) {
+      throw new Error("BackfillSync - Non linear blocks");
+    }
+    if (!verifyBlockSignature(config, state, block)) {
```

Comment:
We should use BLS batch verification to verify all proposer signatures in one go

Reply:
How can you retrieve the pubkey of the block proposer for a block that's pretty old? Do we keep the shuffling somewhere?

Reply:
I don't think we can that's why I'm using this slow method with current state we have